### ISSUE TITLE
Video slowly getting out of sync (slower) than audio and subtitles

### ISSUE THREAD

**Issue by arximboldi** [k4yt3x/video2x#1132]
This is the input (don't judge me):

![image](https://github.com/k4yt3x/video2x/assets/4521138/4686a184-90eb-4ebc-8277-da7b1f98c002)

The generated video slowly gets out of sync (it plays slower than the subtitles and audio, which are in sync relative to eachother).

Not sure what's going on...

**Comment by arximboldi** [comment]
Ok, so here is what `video2x` thinks is the number of frames: 40321

But `nb_read_frames=41076`...
```
ffprobe '[HR] Bible Black - Episode 1 [073E8ABB].mkv' -select_streams v:0 -count_frames -show_entries stream=nb_read_frames 
```

This would explain the difference... we are feeding less frames, and `ffmpeg` is stretching the content to fit the audio length, causing the delay...

**Comment by arximboldi** [comment]
I'm working on a fix...

**Comment by arximboldi** [comment]
Ok, so that was not the issue. I've changed the code to use the frame cound calculated by `ffprobe`... now it generates a video that is longer than the original one. Something's off... why is it reading more frames than it should? Is it interpolating and generating extra frames? Or maybe the frame rate information is wrong? Or is it variable frame rate?

**Comment by arximboldi** [comment]
I found a solution, that also fixes #780 ... I'll make a PR soon.
